frontend/src/app/components/navigation/navigationpills/NavigationPills.tsx

```diff
@@ -1,50 +1,83 @@
-import React, { useEffect, useState } from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import './NavigationPills.css';
 import { INavigationPills } from './INavigationPills';
-import { Link, useLocation } from 'react-router-dom';
+import { useLocation } from 'react-router-dom';
 import Actions from '../../action/Actions';
 import { Button } from '../../button/Button';
 import useMediaQuery from '../../../hooks/useMediaQuery';
 
 const NavigationPills: React.FC<INavigationPills> = ({
-  items,
   components,
-  dropdownItems,
   isDisable,
 }) => {
-  const [activeIndex, setActiveIndex] = useState<number>(0);
+  const [activeTabKey, setactiveTabKey] = useState<string>('');
```

Comment:
would it make sense to have the default state value as `useState<string>(components[0].value)`? This way you can get rid of the useEffect on lines 43-50

Reply:
I tried to set it as default value and works in most of the cases , but in 1 scenario it fails and do not render component. So keeping it same.